projects/graphql-client/src/graphql.module.ts

```diff
@@ -1,29 +1,27 @@
 import { Inject, InjectionToken, Injector, ModuleWithProviders, NgModule } from '@angular/core';
-import { ApolloModule, APOLLO_OPTIONS } from 'apollo-angular';
-import { HttpLink, HttpLinkModule } from 'apollo-angular-link-http';
-import { InMemoryCache, NormalizedCacheObject } from 'apollo-cache-inmemory';
-import { ApolloClientOptions } from 'apollo-client';
-import { GraphQlQueryHandler, GRAPHQL_REQUEST_HANDLERS_TOKENS, GRAPHQL_URI } from './graphql-config';
+import { InMemoryCache } from '@apollo/client/core';
+import { APOLLO_OPTIONS } from 'apollo-angular';
+import { HttpBatchLink } from 'apollo-angular/http';
+import {
+  GraphQlQueryHandler,
+  GRAPHQL_BATCH_SIZE,
+  GRAPHQL_REQUEST_HANDLERS_TOKENS,
+  GRAPHQL_URI
+} from './graphql-config';
 import { GraphQlRequestService } from './graphql-request.service';
 
-// tslint:disable-next-line: only-arrow-functions
-export function createApollo(httpLink: HttpLink, uri: string): ApolloClientOptions<NormalizedCacheObject> {
-  return {
-    link: httpLink.create({ uri: uri }),
-    cache: new InMemoryCache({
-      // Never assume an object with an ID is safe to cache, use the whole request path as the key only
-      dataIdFromObject: () => undefined
-    })
-  };
-}
-
 @NgModule({
-  imports: [HttpLinkModule, ApolloModule],
   providers: [
     {
       provide: APOLLO_OPTIONS,
-      useFactory: createApollo,
-      deps: [HttpLink, GRAPHQL_URI]
+      useFactory: (httpLink: HttpBatchLink, uri: string, batchSize: number) => ({
+        cache: new InMemoryCache(),
```

Comment:
`dataIdFromObject` is deprecated. InMemoryCache now supports Type policies for defining cache strategies. By default, it would use TypeName + data id to store the object and would take into account the request params to determine whether it needs to refetch or it can serve from cache. We can set different policies for each Gql Request. I plan to add that, but not as part of this PR.

For an array, it will use request and type name and  `[]` to serve from cache. So if request changes, previous cache would not be used.

Reply:
We can optimize caching as a later PR as long as things are still working here. Type name and data id is what it originally used which is not sufficient because we request different attributes for the same entity in different requests. I'm certainly ok with this as long as it's working but please pay extra careful attention to making sure requests with different arguments, different selections etc. are all working for the same entity.

Reply:
Yes. So If we request a new attribute and apollo can't find it in the cache for the given object, it would fire a new request, update the cache and then server the response. I have tested few scenarios like this. Didn't find any issues yet.